WMF Framework/Event Platform/EventPlatformClient.swift

```diff
@@ -383,26 +388,53 @@ public class EventPlatformClient: NSObject, SamplingControllerDelegate {
      * fire-and-forget fashion
      */
     private func postAllScheduled(_ completion: (() -> Void)? = nil) {
-        DDLogDebug("EPC: Posting all scheduled requests")
+        guard let storageManager = self.storageManager, let streamConfigs = self.streamConfigurations else {
+            completion?()
+            return
+        }
+
+        DDLogDebug("EPC: Processing all scheduled requests")
+        let events = storageManager.popAll()
+        if events.count == 0 {
+            DDLogDebug("EPC: Nothing to send.")
+            completion?()
+            return
+        }
+
         let group = DispatchGroup()
-        while let item = outputBufferPopFirst() {
+        for event in events {
             group.enter()
-            httpPost(url: item.url, body: item.data) { result in
+
+            guard let streamConfig = streamConfigs[event.stream] else {
+                /// Stream is not configured.
+                DDLogWarn("EPC: Event submitted for stream '\(event.stream.rawValue)' but only the following streams are configured: \(streamConfigs.keys.map(\.rawValue).joined(separator: ", "))")
+                storageManager.markPurgeable(event: event)
+                group.leave()
+                continue
+            }
+
+            guard samplingController.inSample(stream: event.stream, config: streamConfig) else {
+                /// Stream is out of sample. Mark as failed so it will be dropped from storage on the next cleanup.
+                DDLogDebug("EPC: Stream '\(event.stream.rawValue)' is not in sample")
+                storageManager.markPurgeable(event: event)
+                group.leave()
+                continue
+            }
```

Comment:
Are these checks needed here? They already go through this filter in `_submit` and `loadStreamConfiguration` before entering the output buffer. Seems like this check would be pruning against the future config rather than keeping the present config pruning that has already happened. That's more of a backend question for y'all - I could see the argument for enforcing a hard drop against the most up-to-date config.

Reply:
Nice catch.  I think you're right, these last-second rechecks are neither necessary nor desirable.  I'll remove them.